src/vs/editor/browser/controller/mouseHandler.ts

```diff
@@ -407,12 +415,30 @@ class MouseDownOperation extends Disposable {
 		this._dispatchMouse(position, true);
 	}
 
+	private _onMouseDragThenMove(e: EditorMouseEvent): void {
```

Comment:
Merge into `_onMouseDownThenMove` and use `mouseState.lastMouseDownEvent` (which should be renamed to something better, see other comment) to differentiate the two cases